chapter_linear-networks/softmax-regression-scratch.ipynb

```diff
@@ -354,7 +354,11 @@
     "        // before we can use it with NDIndex to \"pick\" indices. \n",
     "        // It also takes in a boolean for returning a copy of the existing NDArray\n",
     "        // but we don't want that so we pass in `false`.\n",
-    "        return yHat.get(new NDIndex(\":, {}\", y.toType(DataType.INT32, false))).log().neg();\n",
+    "        NDIndex pickIndex = new NDIndex()\n",
+    "                 .addAllDim(Math.floorMod(-1, yHat.getShape().dimension()))\n",
+    "                 .addPickDim(y);\n",
+    "        return yHat.get(pickIndex).log().neg();\n",
+    "        // return yHat.get(new NDIndex(\":, {}\", y.toType(DataType.INT32, false))).log().neg();\n",
```

Comment:
```suggestion
```

Reply:
I agree the zachgk's suggestion. Thanks.

Reply:
Cool! But we can't edit your branch. Could you remove it on your side and then we can merge.

Reply:
Thanks for the comment. I have done.